src/App.js

```diff
@@ -1,16 +1,30 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatLog, setChatLog] = useState(chatMessages);
+  const [likeCount, setLikeCount] = useState(0);
+
+  const UpdateLikes = (id, like) => {
+    const currentEntry = chatLog.find((entry) => entry.id === id);
+    currentEntry.liked = like;
+    setChatLog([...chatLog]);
+    if (like) setLikeCount(likeCount + 1);
+    if (!like) setLikeCount(likeCount - 1);
+  };
```

Comment:
This is a very unique way of doing this. We have to be a little careful because you haven't made a copy of the chatLog, so it is possible in a more robust environment you could end up with some unintentional side effects, but it should be fine with what we have.
I usually have been arguing against the need of a likeCount state variable because it can be easily counted, but I think you have used it effectively here!